packages/core-api/src/apis/definitions/StorageApi.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApiRef } from '../ApiRef';
+
+type UnsubscribeFromStore = () => void;
+type SubscribeToStoreHandler<T> = ({
+  storeName,
+  key,
+  oldValue,
+  newValue,
+}: {
+  storeName: string;
+  key: string;
+  oldValue?: T;
+  newValue?: T;
+}) => void;
+
+export type StorageApi = {
+  /**
+   * Get persistent data.
+   * @param {String} storeName Name of the store.
+   * @param {String?} key (Optional) Unique key associated with the data.
+   * If not key is specified,the whole store is returned.
+   * @param {String} key (Optional) Empty value to return if there is not data.
+   * @return {Object} data The data that should is stored.
+   */
+  getFromStore<T>(
```

Comment:
I feel we could keep the method names a bit shorter, store would typically already be part of the name when calling this API.

```ts
storageApi.getFromStore('my-store', 'my-key')
```
vs
```ts
storageApi.get('my-store', 'my-key')
```